source/lib/cdk-textract-stack.ts

```diff
@@ -122,6 +127,276 @@ export class CdkTextractStack extends cdk.Stack {
+          enableKeyRotation: true
+        }
+      );
+
+      const kendraRole = new iam.Role(
+        this,
+        this.resourceName("KendraRole"),
+        {
+          assumedBy: new iam.ServicePrincipal("kendra.amazonaws.com")
+        }
+      );
+
+      kendraRole.assumeRolePolicy?.addStatements( new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ["sts:AssumeRole"],
+        principals: [ new iam.ServicePrincipal("kendra.amazonaws.com") ]
+      })
+      );
+
+      kendraRole.addToPolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["cloudwatch:PutMetricData"],
+          resources: ["*"],
+          conditions: {"StringEquals": {"cloudwatch:namespace": "AWS/Kendra"}} 
+        })
+      );
+        
+      kendraRole.addToPolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["logs:DescribeLogGroups"],
+          resources: ["*"]
+        })
+      );
+
+      kendraRole.addToPolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["logs:CreateLogGroup"],
+          resources: ["arn:aws:logs:"+this.region+":"+this.account+":log-group:/aws/kendra/*"]
+        })
+      );
+
+      kendraRole.addToPolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["logs:DescribeLogStreams", "logs:CreateLogStream", "logs:PutLogEvents"],
+          resources: ["arn:aws:logs:"+this.region+":"+this.account+":log-group:/aws/kendra/*:log-stream:*"]
+        })
+      );
+
+      kendraRole.addToPolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["s3:GetObject", "s3:ListBucket"],
+          resources: [
+            covidDataBucket.bucketArn,
+            `${covidDataBucket.bucketArn}/*`
+          ]
+        })
+      );
+
+      kendraRoleArn = kendraRole.roleArn;
+
+      const onEventCreateKendraIndexLambda = new lambda.Function(this, this.resourceName('OnEventKendraIndexCreator'), {
+        code: lambda.Code.fromAsset('lambda/kendraIndexCreator'),
+        description: 'onEvent handler for creating Kendra index',
+        runtime: lambda.Runtime.PYTHON_3_7,
+        handler: 'lambda_function.lambda_handler',
+        timeout: Duration.minutes(15),
+        environment: {
+          KENDRA_ROLE_ARN: kendraRole.roleArn,
+          KMS_KEY_ID: kendraKMSKey.keyId,
+          KENDRA_INDEX_CLIENT_TOKEN: this.uuid.toLowerCase()
+        }
+      });
+
+      kendraKMSKey.grantEncryptDecrypt(onEventCreateKendraIndexLambda);
+      onEventCreateKendraIndexLambda.addLayers(boto3Layer);
+
+      onEventCreateKendraIndexLambda.addToRolePolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["kendra:CreateIndex","kendra:DeleteIndex","kendra:DescribeIndex","kendra:TagResource","kms:ListKeys",
+          "kms:ListAliases","kms:DescribeKey","kms:CreateGrant"],
```

Comment:
* Please split out out permissions for Kendra and KMS
* Also, do can we not give necessary permissions for using KMS functions -> kendraKMSKey.grant..